src/modules/systemlib/param/param.c

```diff
@@ -576,7 +577,6 @@ param_set_internal(param_t param, const void *val, bool mark_saved, bool notify_
 			}
 
 			memcpy(s->val.p, val, param_size(param));
-			params_changed = true;
```

Comment:
The removal of this line is incorrect - the struct handling is not interpreting a param change and should always return the changed flag as a result.